bot/exts/events/trivianight/_questions.py

```diff
@@ -0,0 +1,240 @@
+
+        self.active_question = False
+
+        return return_dict, answer_embed, time_limit, question_points
+
+
+class Questions:
+    """An interface to use from the TriviaNight cog for questions."""
+
+    def __init__(self, scoreboard: Scoreboard):
+        self.scoreboard = scoreboard
+        self.questions = []
+
+    def set_questions(self, questions: list) -> None:
+        """Setting `self.questions` dynamically via a function to set it."""
+        self.questions = questions
+
+    def next_question(self, number: int = None) -> Union[Embed, None]:
+        """
+        Chooses a random unvisited question from the question bank.
+
+        If the number parameter is specified, it'll head to that specific question.
+        """
+        if all("visited" in question.keys() for question in self.questions) and number is None:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="All of the questions in the question bank have been used.",
+                color=Colours.soft_red
+            )
+
+        if number is None:
+            question_number = randrange(0, len(self.questions))
+            while "visited" in self.questions[question_number].keys():
+                question_number = randrange(0, len(self.questions))
+        else:
+            question_number = number - 1
+
+        self.questions[question_number]["visited"] = True
+
+        # The `self.view` refers to the QuestionView
+        self.view.current_question = self.questions[question_number]
+
+    def list_questions(self) -> Union[Embed, str]:
+        """
+        Lists all questions from the question bank.
+
+        It will put the following into a message:
+            - Question number
+            - Question description
+            - If the question was already 'visited' (displayed)
+        """
+        if not self.questions:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No questions are currently loaded in!",
+                color=Colours.soft_red
+            )
+        spaces = len(
+            sorted(
+                self.questions, key=lambda question: len(question['description'])
+            )[-1]["description"]
+        ) + 3
+        formatted_string = ""
+        for question in self.questions:
+            question_description = question["description"]
+            formatted_string += f"`Q{question['number']}: {question_description}" \
+                                f"{' ' * (spaces - len(question_description) + 2)}" \
+                                f"|` {':x:' if not question.get('visited') else ':white_check_mark:'}\n"
+
+        return formatted_string.strip()
+
+    def current_question(self) -> tuple[Embed, QuestionView]:
+        """Returns an embed entailing the current question as an embed with a view."""
+        return self.view.create_current_question(), self.view
```

Comment:
Could technically be a property no?

Reply:
Hmm, yeah it could be. But, I guess if we're gonna make this a property, we'll have to make other things properties too right?

Reply:
i'm down with changing this to a property though

Reply:
> we'll have to make other things properties too right?

Not have to, it's really not a big deal, just thought I'd bring it up in-case it didn't cross your mind yet.

Reply:
Ah alright. I think we could keep it as is considering most things are functions anyways